mlir/unittests/translation/test_translation.cpp

```diff
@@ -0,0 +1,747 @@
+#include <mlir/Dialect/Func/IR/FuncOps.h>
+#include <mlir/IR/BuiltinOps.h>
+#include <mlir/IR/MLIRContext.h>
+#include <mlir/IR/OwningOpRef.h>
+#include <string>
+#include <utility>
+
+namespace {
+
+class ImportTest : public ::testing::Test {
+protected:
+  std::unique_ptr<mlir::MLIRContext> context;
+
+  void SetUp() override {
+    mlir::DialectRegistry registry;
+    registry.insert<mqt::ir::ref::MQTRefDialect>();
+    registry.insert<mlir::func::FuncDialect>();
+
+    context = std::make_unique<mlir::MLIRContext>();
+    context->appendDialectRegistry(registry);
+    context->loadAllAvailableDialects();
+  }
+
+  void TearDown() override {}
+};
+
+std::string getOutputString(mlir::OwningOpRef<mlir::ModuleOp>* module) {
+  std::string outputString;
+  llvm::raw_string_ostream os(outputString);
+  (*module)->print(os);
+  os.flush();
+  return outputString;
+}
+
+// Adapted from
+// https://github.com/llvm/llvm-project/blob/d2b3e86321eaf954451e0a49534fa654dd67421e/llvm/unittests/MIR/MachineMetadata.cpp#L181
+bool checkOutput(const std::string& checkString,
+                 const std::string& outputString) {
+  auto checkBuffer = llvm::MemoryBuffer::getMemBuffer(checkString, "");
+  auto outputBuffer =
+      llvm::MemoryBuffer::getMemBuffer(outputString, "Output", false);
+
+  llvm::SmallString<4096> checkFileBuffer;
+  const llvm::FileCheckRequest request;
+  llvm::FileCheck fc(request);
+  const llvm::StringRef checkFileText =
+      fc.CanonicalizeFile(*checkBuffer, checkFileBuffer);
+
+  llvm::SourceMgr sm;
+  sm.AddNewSourceBuffer(
+      llvm::MemoryBuffer::getMemBuffer(checkFileText, "CheckFile"),
+      llvm::SMLoc());
+  if (fc.readCheckFile(sm, checkFileText)) {
+    return false;
+  }
+
+  auto outputBufferBuffer = outputBuffer->getBuffer();
+  sm.AddNewSourceBuffer(std::move(outputBuffer), llvm::SMLoc());
+  return fc.checkInput(sm, outputBufferBuffer);
+}
+
+} // namespace
+
+using namespace qc;
+
+TEST_F(ImportTest, EntryPoint) {
+  const qc::QuantumComputation qc(1);
+
+  auto module = translateQuantumComputationToMLIR(context.get(), qc);
+
+  const auto outputString = getOutputString(&module);
+  const auto* checkString = R"(
+    CHECK: func.func @main()
+    CHECK: return
+  )";
+
+  ASSERT_TRUE(checkOutput(checkString, outputString));
+}
+
+TEST_F(ImportTest, Allocation) {
+  const qc::QuantumComputation qc(3);
+
+  auto module = translateQuantumComputationToMLIR(context.get(), qc);
+
+  const auto outputString = getOutputString(&module);
+  const auto* checkString = R"(
+    CHECK: %[[Reg:.*]] = "mqtref.allocQubitRegister"() <{size_attr = 3 : i64}> : () -> !mqtref.QubitRegister
+    CHECK: "mqtref.extractQubit"(%[[Reg]]) <{index_attr = 0 : i64}> : (!mqtref.QubitRegister) -> !mqtref.Qubit
+    CHECK: "mqtref.extractQubit"(%[[Reg]]) <{index_attr = 1 : i64}> : (!mqtref.QubitRegister) -> !mqtref.Qubit
+    CHECK: "mqtref.extractQubit"(%[[Reg]]) <{index_attr = 2 : i64}> : (!mqtref.QubitRegister) -> !mqtref.Qubit
```

Comment:
I only now noticed this while carefully reading through the tests. The current setup is missing one crucial thing and that is the deallocation of the qubit register just before the return statement.
This could be considered boilerplate code, but it is essential in my opinion to form a valid and clean program.